pynucastro/networks/rate_collection.py

```diff
@@ -159,11 +178,41 @@ def eval_abar(self):
             xvec[i] = self.X[n]
         return 1. / np.sum(xvec / avec)
 
-    def __str__(self):
-        ostr = ""
-        for k in self.X:
-            ostr += f"  X({k}) : {self.X[k]}\n"
-        return ostr
+    def bin_as(self, nuclei, *, verbose=False):
+        """given a list of nuclei, return a new Composition object with the
+        current composition mass fractions binned into the new nuclei."""
+
+        # sort the input nuclei by A, then Z
+        nuclei.sort(key=lambda n: (n.A, n.Z))
+
+        # create the new composition
+        new_comp = Composition(nuclei)
+
+        # loop over our original nuclei.  Find the new nucleus such
+        # that n_orig.A >= n_new.A.  If there are multiple, then do
+        # the same for Z
+        for old_n, v in self.X.items():
+
+            candidates = [q for q in nuclei if old_n.A >= q.A]
+            # if candidates is empty, then all of the nuclei are heavier than
+            # old_n, so just put its composition in the first new nucleus
+            # (which will be the lightest)
+            if not candidates:
+                match_nuc = nuclei[0]
+            else:
+                max_A = max(q.A for q in candidates)
+                match_A = [q for q in candidates if q.A == max_A]
+                if len(match_A) > 1:
+                    match_Z = [q for q in match_A.sort(lambda p: p.Z) if old_n.Z >= q]
```

Comment:
`list.sort()` does an in-place sort and returns `None`. We should probably add a test case for this branch.
```suggestion
                    match_Z = [q for q in sorted(match_A, key=lambda p: p.Z) if old_n.Z >= q]
```